marimo/_plugins/ui/_core/ui_element.py

```diff
@@ -269,6 +277,27 @@ def value(self) -> T:
             )
         return self._value
 
+    @value.setter
+    def value(self, value: T) -> None:
+        del value
+        raise RuntimeError(
+            "Setting the value of a UIElement is not allowed. "
+            "If you need to imperatively set the value of a UIElement, "
+            "consider using mo.state()."
+        )
+
+    @property
+    def on_change(self) -> Optional[Callable[[T], None]]:
+        return self._on_change
+
+    @on_change.setter
+    def on_change(self, on_change: Optional[Callable[[T], None]]) -> None:
+        del on_change
+        raise RuntimeError(
+            "Setting the on_change handler of a UIElement is not allowed. "
+            "You must set the on_change in the constructor."
+        )
+
```

Comment:
I feel like we don't need these, since .on_change doesn't exist UIElement.

Reply:
Instead, this could be:

- remove the property
- add

```python
def __setattr__(self, name: str, value: Any) -> None:
  if name == "on_change":
        raise RuntimeError(
          "Setting the on_change handler of a UIElement is not allowed. "
          "You must set the on_change in the constructor."
      )
  super().__setattr__(name, value)
```